parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/HadoopPositionOutputStream.java

```diff
@@ -62,7 +64,9 @@ public void flush() throws IOException {
   @Override
   public void close() throws IOException {
     try (FSDataOutputStream fdos = wrapped) {
-      fdos.hflush();
+      if (fdos.hasCapability("hflush")) {
```

Comment:
1. adding a delegating `hasCapability()` might be handy
2. consider what to do on a sync failure. close() will sill be invoked but any exception thrown would probably be from the hflush() failure.